x/ecocredit/simulation/basket/utils.go

```diff
@@ -0,0 +1,105 @@
+package basketsims
+
+import (
+	"math/rand"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	regentypes "github.com/regen-network/regen-ledger/types"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+func GetAndShuffleClasses(sdkCtx sdk.Context, r *rand.Rand, qryClient ecocredit.QueryClient) ([]*ecocredit.ClassInfo, error) {
+	ctx := regentypes.Context{Context: sdkCtx}
+	res, err := qryClient.Classes(ctx, &ecocredit.QueryClassesRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	classes := res.Classes
+	if len(classes) <= 1 {
+		return classes, nil
+	}
+
+	r.Shuffle(len(classes), func(i, j int) { classes[i], classes[j] = classes[j], classes[i] })
+	return classes, nil
+}
+
+func min(x, y int) int {
+	if x > y {
+		return y
+	}
+	return x
+}
+
+func randomExponent(r *rand.Rand, precision uint32) uint32 {
+	exponents := []uint32{0, 1, 2, 3, 6, 9, 12, 15, 18, 21, 24}
+	for {
+		x := exponents[r.Intn(len(exponents))]
+		if x > precision {
+			return x
+		}
+	}
+}
+
+func Contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+// GenAndDeliverTxWithRandFees generates a transaction with a random fee and delivers it.
+func GenAndDeliverTxWithRandFees(txCtx simulation.OperationInput) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
```

Comment:
why is this in x/ecocredit/simulation/basket/utils? shouldn't we just have some x/ecocredit/simulation/utils?

Reply:
It looks this was done to avoid issues with import cycles. The simulation package already imports `WeightedOperations` from the basket simulation package.

Reply:
It looks this was done to avoid issues with import cycles. The simulation package already imports `WeightedOperations` from the basket simulation package.

Reply:
It looks this was done to avoid issues with import cycles. The simulation package imports `WeightedOperations` from the basket simulation package.

Reply:
moving this to `x/ecocredit/simulation/utils` shouldn't cause any import cycle anyway